js/src/index.js

```diff
@@ -553,6 +595,138 @@ class JupyterScatterView {
+    );
+
+    if (TOOLTIP_ALL_VISUAL_CONTENTS.has(content)) {
+      this[`tooltipContent${capitalContent}ChannelName`] = createElementWithClass(
+        'div',
+        ['channel-name', `${content}-channel-name`]
+      );
+      this[`tooltipContent${capitalContent}ChannelName`].textContent = TOOLTIP_ALL_VISUAL_CONTENTS.get(content);
+
+      this[`tooltipContent${capitalContent}Channel`].appendChild(
+        this[`tooltipContent${capitalContent}ChannelName`]
+      );
+
+      this[`tooltipContent${capitalContent}ChannelBadge`] = createElementWithClass(
+        'div',
+        ['channel-badge', `${content}-value-badge`]
+      );
+      this[`tooltipContent${capitalContent}ChannelBadgeMark`] = createElementWithClass(
+        'div',
+        ['channel-badge-mark', `${content}-channel-badge-mark`]
+      );
+      this[`tooltipContent${capitalContent}ChannelBadgeBg`] = createElementWithClass(
+        'div',
+        ['channel-badge-bg', `${content}-channel-badge-bg`]
+      );
+      this[`tooltipContent${capitalContent}ChannelBadge`].appendChild(
+        this[`tooltipContent${capitalContent}ChannelBadgeMark`]
+      );
+      this[`tooltipContent${capitalContent}ChannelBadge`].appendChild(
+        this[`tooltipContent${capitalContent}ChannelBadgeBg`]
+      );
+      this[`tooltipContent${capitalContent}Channel`].appendChild(
+        this[`tooltipContent${capitalContent}ChannelBadge`]
+      );
+    }
+
+    this[`tooltipContent${capitalContent}Value`].appendChild(
+      this[`tooltipContent${capitalContent}ValueText`]
+    );
+
+    this[`tooltipContent${capitalContent}ValueHistogram`] = createHistogram(
+      TOOLTIP_HISTOGRAM_WIDTH,
+      TOOLTIP_HISTOGRAM_HEIGHT
+    );
+    this[`tooltipContent${capitalContent}Value`].appendChild(
+      this[`tooltipContent${capitalContent}ValueHistogram`].element
+    );
+
+    const histogram = this.model.get(`${content}_histogram`) || this.model.get('tooltip_contents_non_visual_info')[content]?.histogram;
+    const scale = this.model.get(`${content}_scale`) || this.model.get('tooltip_contents_non_visual_info')[content]?.scale
+
+    this[`tooltipContent${capitalContent}ValueHistogram`].init(
+      histogram,
+      scale === 'categorical',
+    );
+
+    this[`tooltipContent${capitalContent}Title`].textContent = toTitleCase(
+      this.model.get(`${content}_title`) || content || ''
+    );
+
+    this.tooltipContent.appendChild(this[`tooltipContent${capitalContent}Channel`]);
+    this.tooltipContent.appendChild(this[`tooltipContent${capitalContent}Title`]);
+    this.tooltipContent.appendChild(this[`tooltipContent${capitalContent}Value`]);
+  }
+
+  createTooltipContentsDomElements() {
+    this.tooltipContent.replaceChildren();
+
+    const contents = new Set(this.tooltipContentsAll);
+
+    // To ensure that visual contents are listed before non-visual contents,
+    // we're first going to iterate over the visual contents.
+    for (const content of TOOLTIP_ALL_VISUAL_CONTENTS) {
```

Comment:
Think this is probably what we want?

```suggestion
    for (const content of TOOLTIP_ALL_VISUAL_CONTENTS.keys()) {
```

Reply:
See my other comment. I think we want to remove this whole loop